src/main/java/htsjdk/samtools/cram/build/ContainerFactory.java

```diff
@@ -58,8 +58,7 @@ public Container buildContainer(final List<CramCompressionRecord> records) {
             slices.add(slice);
         }
 
-        final Container container = Container.initializeFromSlices(slices);
-        container.compressionHeader = compressionHeader;
+        final Container container = Container.initializeFromSlices(slices, compressionHeader);
```

Comment:
We need the compressionHeader when we call `getCraiEntries()` so requiring it's here helps ensure it's valid